src/main/java/it/gov/pagopa/rtp/sender/utils/DateUtils.java

```
@@ -26,4 +30,29 @@ public static String localDateTimeToCustomOffsetFormat(LocalDateTime localDateTi
         .orElseThrow(() ->
             new IllegalArgumentException("Couldn't convert local datetime to offset format"));
   }
+
+  /**
+   * Converts a timestamp expressed in microseconds since epoch to a LocalDate
+   * in the "Europe/Rome" timezone. The input is divided by 1000 to convert it to milliseconds,
+   * then transformed into a LocalDate, discarding the time component.
+   *
+   * @param timestamp the timestamp in microseconds (e.g. 1753866547153000)
+   * @return the corresponding LocalDate in Europe/Rome timezone, or null if the input is invalid
+   */
+  public static LocalDate convertLongToLocalDate(Long timestamp) {
+    try {
+      long millis = (timestamp > 10000000000000L) ? timestamp / 1000 : timestamp;
+
+      LocalDate result = Instant.ofEpochMilli(millis)
+          .atZone(ZoneId.of("Europe/Rome"))
```

If the try-catch fails, we'll return null. Is this okay? Or will it result in a null pointer exception?

---

It shouldn't result in a null pointer because the field we map doesn't have any restraint, and this doesn't bring problems in the DELETE case, at least for now(until we come up with another solution for the delete case) it's ok

---

Ok, I was thinking about that too. It shouldn't result in a null pointer because the mapped field has no constraint, but we could throw an exception and block the creation entirely, to avoid generating RTPs without an expiryDate